test/java/seedu/addressbook/util/TypicalPersons.java

```diff
@@ -0,0 +1,57 @@
+package seedu.addressbook.util;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.UniquePersonList;
+
+/**
+ * Class to generate typical test persons
+ */
+public class TypicalPersons {
+
+    public static Person amy, bill, candy, donald, grace;
```

Comment:
@pyokagan  I agree. But I don't mind doing it as a separate issue. The extra work caused by duplication of code is a good source of 'low hanging fruit' issues for new contributors to do. 

On a somewhat related note, at the moment I'm not too eager to unify similarities between the levels. At our final steady state it may be desirable to have different levels doing the same thing in different ways, so that students get to see more ways of doing the same thing. As we are not yet certain what our final steady state is going to be, I don't mind if levels evolved somewhat independently for now.

In hindsight, I'm not sure if we should use the Builder pattern in level 2. It would be good for one level (level 2?) to not use the Builder pattern so that it can be contrasted with the next level (level 3?) that improves the situation by applying the Builder pattern. Anyway, that's something we can consider on a future day. We can leave it be for now. :-p